keps/sig-auth/20191204-x509-certificate-for-serviceaccount.md

```diff
@@ -0,0 +1,286 @@
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an issue in [kubernetes/enhancements] referencing this KEP and targeting a release milestone **before [Enhancement Freeze](https://github.com/kubernetes/sig-release/tree/master/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This KEP describes Kubernetes generates and mounts ServiceAccounts' x509 certificates for Pods, 
+and Kubernetes can get ServiceAccount identity information if Pods visit Kubernetes using ServiceAccounts' x509 certificates.
+
+## Background
+
+By default, Pod running in Kubernetes will be mount Kubernetes root CA and ServiceAccount JWT(which will be mount at path `/var/run/secrets/kubernetes.io/serviceaccount`). 
+And Kubernetes can get ServiceAccount identity information if Pod visit Kubernetes using SerivceAccount JWT(common usage: `client-go`), 
+then Kubernetes will do Authz according the ServiceAccount identity information and the requested resources.
+
+## Motivation
```

Comment:
I'm not sold on the scalability aspect of this. If you want motivation, checkout: https://github.com/kubernetes/kubernetes/issues/63732

Reply:
I'm not sure if the latest codes also has this situation:

10k Pod using a deleted ServiceAccountToken to visit API Server, and API Server need to visit etcd 10k times to check if Token is exist and valid, because our cachedTokenAuthenticator failureTTL is 0.